### ISSUE TITLE
Add requirements for Javascript code contributions

### ISSUE THREAD

**Issue by s-ben** [decred/dcrdocs#862]
As discussed in this [Issue](https://github.com/xaur/decred-issues/issues/115), a JavaScript library has been requested multiple times. In a recent [discussion](https://riot.im/app/#/room/!MIGqWXfLFBwhipPKYL:decred.org/$15500054718437GyoNW:decred.org), there appeared to be rough consensus around the need for this library (e.g. to support JS-based wallets), as well as the requirements of JS code generally. 

Per @davecgh 's suggestion, the library should be written in TypeScript (as defined [here](https://www.typescriptlang.org/)).

@oregonisaac has created a gist ([Decred Open-Source JS Library Requirements](https://gist.github.com/oregonisaac/e41826e171b53826ff869cab75d8ed0b)) which collects the functional requirements of such a library, as well as research on best practices. He also suggests forking the Go-based contribution requirements for other languages. 

I suggest adding a Javascript section to the [Source Code Contributions](https://docs.decred.org/contributing/source-code-contributions/) page, alongside the Go requirements.

**Comment by xaur** [comment]
Side note: this is getting into the 'dev docs' territory. I remember user `nwops` and someone else were going to build dev docs website, but that didn't gain much traction.

**Comment by s-ben** [comment]
Yeah, that was me and `nwops`. He was spearheading the project when I joined back in September, and I was looking to contribute to it. He was working on several other projects though, and didn't really have the bandwidth to push it forward. Then afaik, he got some other paid gigs and moved on.  Haven't seen him on here in a few months. He did put together a skeleton of a hugo site for dev docs, which he was hosting on his gitlab. After he left, I forked it to a gitlab account of my own, with the idea of continuing it. Here's the hosted version of what we have if you're curious.
https://s_ben.gitlab.io/decred-dev-docs/docs/ 

I've been looking at taking this up again myself. But while the subject occasionally comes up, when I've tried to gauge community interest in it (i.e. pinging devs for interest in collaboration, talking about it on slack), I've not gotten much interest. It's also not clear to me how much we need them at this point. The documentation on the individual repos is fairly fleshed out and decentralized (different repos can have their own rules and norms around GitHub process, for instance), and basic technical docs on important subjects can be found in dcrdocs. So the general strategy I've adopted is to just keep my eye out for holes in the technical documentation, and proactively create pages on dcrdocs that we could port to a potential dev docs should we create it. For instance, I recently created the 'Address Details' page, as that is an important reference doc,
https://docs.decred.org/advanced/address-details/

I still have my eye on doing a full separate dev docs site, but it's a pretty big project (like, could be a couple months of full-time work). So am generally holding off until the community asks for it. Working on that by myself, and annoying busy devs with questions and feedback requests for months on end, without signaling from the community that it's a need, does not sound fun. If we get to a point where we basically have created all the 'raw material' technical content in dcrdocs, creating a separate dev docs could be quicker and more fun.

**Comment by xaur** [comment]
Thanks for background. Good points, dev pages can be later moved to dev docs when there is such a need.
